File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        int b = 0, e = nums.size();

        if (nums[0] < nums[e - 1]) {
            return nums[0];
        }
        
        while (b < e) {
            int m = (b + e) / 2;
            if (nums[0] < nums[m]) {
                b = m + 1;
            } else {
                if (1 == m) {
                    return nums[m];
                }
                
                e = m;
            }
        }

        return nums[b];
    }
};

File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        begin = 0
        end = len(nums) - 1
        while begin < end and nums[begin] > nums[end]:
            middle = (begin + end) // 2
            if nums[begin] <= nums[middle]:
                begin = middle + 1
            else:
                end = middle

        return nums[begin]

        